app/models/app_feature_completion.rb

```diff
@@ -7,12 +7,9 @@ class AppFeatureCompletion < ActiveRecord::Base
   validates :app, :split, :version, presence: true
   validates :split, uniqueness: { scope: :app }
 
-  # This scope requires you to BYO `splits` FROM clause either via join or an
-  # outer scope, if using this as a subselect.
```

Comment:
I realized this was a factoring problem - I can inject the dependency on splits at the point of use when merging into a scope on Split and make this a regular scope. Followed the same pattern on AppRemoteKill scope.